File: myclient/src/app/models/order.ts
import { Address } from "./Address";

export interface ShippingAddress extends Address {  }

export interface OrderItem {
    id: number;
    name: string;
    pictureUrl: string;
    price: number;
    quantity: number;
}

export interface Order {
    id: number;
    buyer: string;
    orderDate: string;
    shipToAddress: ShippingAddress;
    orderItems: OrderItem[];
    subtotal: number;
    orderStatus: string;
    shippingPrice: number;
    total: number;
}

export interface OrderParams {
    pageNumber: number;
    pageSize: number;
    dateFrom?: string;
    dateTo?: string;
}

File: MyAPI/Core/Entities/OrderAggregate/Order.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core.Entities.OrderAggregate
{
    public class Order : BaseEntity
    {
        public Order(){}

        public Order(IReadOnlyList<OrderItem> orderItems, string buyer, OrderAddress shippingAddress,
            DeliveryMethod deliveryMethod, double subtotal, string paymentIntentId)
        {
            Buyer = buyer;
            ShipToAddress = shippingAddress;
            DeliveryMethod = deliveryMethod;
            OrderItems = orderItems;
            Subtotal = subtotal;
            PaymentIntentId = paymentIntentId;
        }

        public string Buyer { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        public OrderAddress ShipToAddress { get; set; }
        public DeliveryMethod DeliveryMethod { get; set; }
        public IReadOnlyList<OrderItem> OrderItems { get; set; }
        public double Subtotal { get; set; }
        public OrderStatus Status { get; set; } = OrderStatus.Pending;
        public string PaymentIntentId { get; set; }
        public bool IsDelete { get; set; }
        public double GetTotal()
        {
            return Subtotal + DeliveryMethod.Price;
        }
    }
}
